### PR TITLE
Cythonize  splitting algorithm to reduce overhead.

### PR DESCRIPTION
**Pull request by rhpvorderman**
### Checklist
- [x] Pull request details were added to CHANGELOG.rst

### CODE REVIEW

File: src/fastqsplitter/split.pyx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# cython: language_level=3
+
+
+import io
+from typing import List
+
+
+def filesplitter(input_handle: io.BufferedReader,
+                 output_handles: List[io.BufferedWriter],
+                 lines_per_block=100):
```
Comment (DavyCats) [inline src/fastqsplitter/split.pyx:32]: A single fastq record is at least four lines, maybe I'm overlooking something but if I were to set this to some value which is not divisible by 4, I would end up with invalid fastq files, right?

Maybe one more check to make sure lines_per_block is divisible by 4 should be added. (And a comment stating that multi-line fastq records are not supported)

File: src/fastqsplitter/split.pyx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            i = 0
+        # This works, if blocksize == 100. Then i will be [0, 1, 2, .., 98, 99]
+        # which is exactly 100 numbers.
+
+    # Write remainder to file. Since stuff is only written at i == blocksize
+    # there is a remainder that needs to be written.
+    output_handles[group_no].write(b"".join(block))
+
+    # The resetting of i at blocksize accomplishes two things:
+    # 1. i will never be larger than blocksize. We do not have to keep
```
Comment (DavyCats) [inline src/fastqsplitter/split.pyx]:
```suggestion
    # 1. it will never be larger than blocksize. We do not have to keep
```

**Review by DavyCats — CHANGES_REQUESTED** [review]

File: src/fastqsplitter/split.pyx (hunk 1 above)
Comment (rhpvorderman) [reply src/fastqsplitter/split.pyx:32]: This is a filesplitter. Thus it is not aware of the fastq format.
The fastq format is handled upstream. Where the group_size is multiplied by  4. This results in a number that is always divisible by 4.

Comment (rhpvorderman) [reply src/fastqsplitter/split.pyx:32]: https://github.com/LUMC/fastqsplitter/blob/def0c115e1165e79e71856672e1f9515095ede7b/src/fastqsplitter/__init__.py#L108

Comment (rhpvorderman) [reply src/fastqsplitter/split.pyx:32]: > This application does NOT work with multiline fastq sequences.

Is stated in the README.

Comment (DavyCats) [reply src/fastqsplitter/split.pyx:32]: :+1: nevermind :stuck_out_tongue:

**Review by DavyCats — APPROVED** [review]

**Pull request merged**